rosbag2_storage_mcap/test/rosbag2_storage_mcap/test_mcap_storage.cpp

```diff
@@ -118,6 +118,9 @@ TEST_F(TemporaryDirectoryFixture, can_write_and_read_basic_mcap_file)
                   1u}}));
     EXPECT_THAT(metadata.message_count, Eq(1u));
 
+    const auto current_distro = rcpputils::get_env_var("ROS_DISTRO");
+    EXPECT_EQ(metadata.ros_distro, current_distro);
```

Comment:
@emersonknapp Am I understand correctly that in `Humble` and `Iron` backport it will be added `std::string get_recorded_ros_distro() const;` API in the `rosbag2_storage_mcap` ?

Reply:
`get_recorded_ros_distro` is not part of the `ReadWriteInterface` that readers/writers use to interact with the plugin. Can we even backport an update to the `BagMetadata` structure? This change feels un-backportable, and the fact that there is a function on `sqlite_storage` doesn't seem like it actually does anything in Humble and Iron, unless I'm missing something - there is nothing that can call it.

Reply:
The one thing that I think I could do with no problem is backport the change to add the ROS_DISTRO metadata in the mcap. That way, newer versions of rosbag2 could read the value out and understand it.

Reply:
@emersonknapp Right we can't add new member in BagMetadata structure on Humble and Iron.
And as regards to the 
> and the fact that there is a function on sqlite_storage doesn't seem like it actually does anything in Humble and Iron, unless I'm missing something - there is nothing that can call it.

Though the `std::string get_recorded_ros_distro() const;` really does things by getting ros distro from the sqlite storage plugin. For `Humble` and `Iron` we don't expect that `get_recorded_ros_distro()` will be used somewhere else rather than in tests. 
However, we need to store ros distro name in mcap and sqlite3 storages on those ros2 distros to be able to parse it in third-party tools like `Foxglove Studio` and future ros2 releases for backward compatibility to be able to determine on the latter releases (Rolling or "Jazy") that some bag was recorded on Humble or Iron.